### PR TITLE
Remove deprecated Alluxio metastore

### PR DESCRIPTION
**Pull request by apc999**
## Description

Remove deprecated Alluxio metastore.
This is a follow up for https://github.com/trinodb/trino/pull/13559/

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by beinan — APPROVED** [review]
lgtm, many thanks for removing this deprecated feature!

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastoreModule.java [hunk 1]
```diff
@@ -50,10 +50,6 @@ protected void setup(Binder binder)
             bindMetastoreModule("thrift", new ThriftMetastoreModule());
             bindMetastoreModule("file", new FileMetastoreModule());
             bindMetastoreModule("glue", new GlueMetastoreModule());
-            // Load Alluxio metastore support through reflection. This makes Alluxio effectively an optional dependency
-            // and allows deploying Trino without the Alluxio jar. Can be useful if the integration is unused and is flagged
-            // by a security scanner.
-            bindMetastoreModule("alluxio-deprecated", deferredModule("io.trino.plugin.hive.metastore.alluxio.AlluxioMetastoreModule"));
```
Comment (ksobolew) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastoreModule.java:56]: You can also remove the `deferredModule` method, it was added specifically for Alluxio and I don't hing we'll need it any more.

**Review by ksobolew — APPROVED** [review]

**Comment by apc999** [comment]
it looks like some UT is flaky. @ksobolew can you help re-run the failing tests?

File: plugin/trino-session-property-managers/pom.xml [hunk 2]
```diff
@@ -155,25 +155,13 @@
             <artifactId>trino-hive</artifactId>
             <type>jar</type>
             <scope>test</scope>
-            <exclusions>
```
Comment (ebyhr) [inline plugin/trino-session-property-managers/pom.xml:158]: Please squash commits into one.

Comment (apc999) [reply plugin/trino-session-property-managers/pom.xml:158]: done

**Comment by apc999** [comment]
@ksobolew @ebyhr can you help merge this PR? thanks!

**Comment by apc999** [comment]
@ebyhr kindly request your assistance in merging this?

**Comment by apc999** [comment]
@mosabua  can you help take a look at this PR and merge it?

**Review by mosabua — APPROVED** [review]
Thank you for that clean up. Everything passes and looks good as well. I will merge.

**Comment by mosabua** [comment]
fyi @colebow .. I think this does need a release notes entry

**Pull request merged**